### PR TITLE
Add e2e test for instruction type `goto_table`

### PR DESCRIPTION
**Pull request by gretelliz**
This is related to PR https://github.com/kytos-ng/sdntrace_cp/pull/108.

### Summary

Add cases for intra-EVC and inter-EVCs with flows with `goto_table`.
Add the [`amlight_looped`](https://github.com/kytos-ng/kytos-end-to-end-tests/blob/fb5e94719e3133e86206d60962e9a640c5ed7d47/tests/helpers.py#L81-L123) topology as a subclass of `AmlightTopo` in `helpers.py`. This add 2 loops in the switches `Amlight1` and `Amlight4`.
Add `test_e2e_42_sdntrace.py` that uses `amlight_looped`.

### Local Tests

```
+ python3 -m pytest --timeout=180 tests/test_e2e_42_sdntrace.py
============================= test session starts ==============================
platform linux -- Python 3.9.2, pytest-7.2.0, pluggy-1.3.0
rootdir: /tests
plugins: rerunfailures-10.2, timeout-2.1.0, anyio-3.6.2
timeout: 180.0s
timeout method: signal
timeout func_only: False
collected 2 items

tests/test_e2e_42_sdntrace.py ..                                         [100%]

=============================== warnings summary ===============================
test_e2e_42_sdntrace.py: 84 warnings
  /usr/lib/python3/dist-packages/mininet/node.py:1121: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    return ( StrictVersion( cls.OVSVersion ) <

test_e2e_42_sdntrace.py: 84 warnings
  /usr/lib/python3/dist-packages/mininet/node.py:1122: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    StrictVersion( '1.10' ) )

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
------------------------------- start/stop times -------------------------------
================= 2 passed, 168 warnings in 158.00s (0:02:38) ==================
```

### End-to-End Tests

N/A

### CODE REVIEW

File: tests/test_e2e_42_sdntrace.py [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+    @staticmethod
+    def get_evc(circuit_id):
+        api_url = KYTOS_API + '/kytos/mef_eline/v2/evc/'
+        response = requests.get(api_url+circuit_id)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        return data
+
+    @classmethod
+    def wait_until_evc_is_active(
```
[the remark this answers is not in the document]
Comment (gretelliz) [reply tests/test_e2e_42_sdntrace.py:66]: You are right. Removed.

File: tests/test_e2e_42_sdntrace.py [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+        if evc["active"]:
+            return evc
+        time.sleep(wait_secs)
+        if i < max_i:
+            return cls.wait_until_evc_is_active(evc_id, wait_secs, i + 1, max_i)
+        else:
+            raise ValueError(f"TimeoutError: {evc_id} didn't get active. {evc}")
+
+    def test_trace_goto_table_intra(cls):
+        """Test trace rest call."""
```
[the remark this answers is not in the document]
Comment (gretelliz) [reply tests/test_e2e_42_sdntrace.py:80]: Done!

File: tests/test_e2e_42_sdntrace.py [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+        assert result[0]['port'] == 16
+        assert result[0]['vlan'] == 200
+
+        assert result[1]['port'] == 18
+        assert result[0]['vlan'] == 200
+        assert result[1]['out']['port'] == 15
+        assert result[1]['out']['vlan'] == 201
+
+    def test_trace_goto_table_inter(cls):
+        """Test trace rest call."""
```
[the remark this answers is not in the document]
Comment (gretelliz) [reply tests/test_e2e_42_sdntrace.py:268]: Done!

File: tests/test_e2e_42_sdntrace.py [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+            return evc
+        time.sleep(wait_secs)
+        if i < max_i:
+            return cls.wait_until_evc_is_active(evc_id, wait_secs, i + 1, max_i)
+        else:
+            raise ValueError(f"TimeoutError: {evc_id} didn't get active. {evc}")
+
+    def test_trace_goto_table_intra(cls):
+        """Test trace rest call."""
+        circuit_id = cls.create_evc(201, interface_a="00:00:00:00:00:00:00:01:15", interface_z="00:00:00:00:00:00:00:01:16")
```
[the remark this answers is not in the document]
Comment (gretelliz) [reply tests/test_e2e_42_sdntrace.py:81]: @viniarck, you're right, there's no need. I removed it and the tests continue to pass.

File: tests/test_e2e_42_sdntrace.py [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+        assert result[0]['vlan'] == 200
+
+        assert result[1]['port'] == 18
+        assert result[0]['vlan'] == 200
+        assert result[1]['out']['port'] == 15
+        assert result[1]['out']['vlan'] == 201
+
+    def test_trace_goto_table_inter(cls):
+        """Test trace rest call."""
+        circuit_id = cls.create_evc(100)
```
[the remark this answers is not in the document]
Comment (gretelliz) [reply tests/test_e2e_42_sdntrace.py:269]: Done!

File: tests/test_e2e_42_sdntrace.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (viniarck) [reply tests/test_e2e_42_sdntrace.py:81]: Right. Cool. Once you push the commits upstream I'll review again and approve the PR.

File: tests/helpers.py [hunk 6]
```diff
@@ -78,6 +78,50 @@ def build(self):
         self.addLink(h14, Ampath2, port1=1, port2=63)
         self.addLink(h15, AndesLight2, port1=1, port2=64)
 
+class AmlightLoopedTopo(Topo):
```
[the remark this answers is not in the document]
Comment (gretelliz) [reply tests/helpers.py:81]: Done as you suggested.

**Review by viniarck — APPROVED** [review]
LGTM. Feel free to merge by Wed if tests are passing after the refactoring @gretelliz. 

Let's leave it a few days here in case Italo or anybody else wants to also review.

**Pull request merged**